packages/compiler/src/server/completion.ts

```diff
@@ -34,6 +41,7 @@ import { getSymbolDetails } from "./type-details.js";
 export type CompletionContext = {
   program: Program;
   params: CompletionParams;
+  document: TextDocument;
```

Comment:
remove this you should have all this info on `file.file`

Reply:
this is for calling "context.document.offsetAt(context.params.position);" in addModelCompletion, do we have a way to get that info from file.file?